src/ModalBase/ModalCloseButton.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from '..';
+import ModalContext from './ModalContext';
+
+const ModalCloseButton = React.forwardRef(({ as, children, ...props }, ref) => {
+  const { close } = useContext(ModalContext);
+  const type = as;
```

Comment:
There a reason to assign `as` to `type` instead of just passing it through to React.createElement directly?

Reply:
Clarity for those reading is all. happy not to assign it if below is still clear

Reply:
I don't feel strongly. :)